generators/firechip/src/main/scala/TargetMixins.scala

```diff
@@ -72,3 +78,15 @@ trait CanHaveMultiCycleRegfileImp {
   }
 }
 
+trait HasFireSimClockingImp extends HasAdditionalClocks {
+  val outer: HasTiles
+  val (tileClock, tileReset) = p(FireSimClockKey).additionalClocks.headOption match {
+    case Some(RationalClock(_, numer, denom)) if numer != denom => (clocks(1), ResetCatchAndSync(clocks(1), reset.toBool))
```

Comment:
Does this check really do anything? I.e. are there generators that do this math and don't realize they are making a synchronized design?
Also are the magic numbers necessary?

Reply:
The numer != denom check? No, but it saves the catch and sync hardware in the synchronous case. I can remove it. 

I think how we plumb the clocks through the target is mostly a separate question to how we want to expose the generation of those clocks for firesim. I mostly just hacked together the prior to get something working quick.

In light of @jerryz123's recent push to harmonize the targets across firechip and chipyard, we should consider making our targets support multiclock by default. So for RTL bound for target-level RTL simulation or physical design, we'll need to define the other derived clocks somehow. 

To remove the vector-indexing hack here i think what we want in the interim is to generate some clock-map [Key/String, ClockInfo] in the parameters object.  We can provide some sugar to let the generator index the right element in the clock vector using the same key.